pkg/controller/node/nodecontroller.go

```diff
@@ -532,6 +532,32 @@ func (nc *NodeController) tryUpdateNodeStatus(node *api.Node) (time.Duration, ap
 				readyCondition.LastTransitionTime = nc.now()
 			}
 		}
+
+		// Like NodeReady condition, NodeOutOfDisk was last set longer ago than gracePeriod, so update
+		// it to Unknown (regardless of its current value) in the master, without contacting kubelet.
+		oodCondition := nc.getCondition(&node.Status, api.NodeOutOfDisk)
+		if oodCondition == nil {
+			glog.V(2).Infof("Out of disk condition of node %v is never updated by kubelet", node.Name)
+			oodCondition := &api.NodeCondition{
```

Comment:
nit: Is there any reason to do it this way instead of the way it was done for readyCondition, namely
node.Status.Conditions = append(node.Status.Conditions, api.NodeCondition{ ...}) ?
If not, I'd suggesting mirroring the way it was done for readyCondition, just to make it clearer to the reader that what you did is exactly the same logic.